pkg/collector/corechecks/containerimage/processor.go

```diff
@@ -115,12 +127,31 @@ func (p *processor) processImage(img *workloadmeta.ContainerImageMetadata) {
 			}
 		}
 
+		// Because we split a single image entity into different payloads if it has several repo digests,
+		// me must re-compute `image_name`, `short_image` and `image_tag` tags.
+		ddTags2 := make([]string, 0, len(ddTags))
+		for _, ddTag := range ddTags {
+			if !strings.HasPrefix(ddTag, "image_name:") &&
+				!strings.HasPrefix(ddTag, "short_image:") &&
+				!strings.HasPrefix(ddTag, "image_tag:") {
+				ddTags2 = append(ddTags2, ddTag)
+			}
+		}
+
+		ddTags2 = append(ddTags2,
+			"image_name:"+repo,
+			"short_image:"+shortName)
```

Comment:
can we also generate the `image_id` tag? that correspond to `image_id:<image_name>@<repoDigests>`

Reply:
Indeed, as it depends on the `repoDigests` that we manipulate in the check, compared to what’s in `workloadmeta` and the `tagger`, it needs to be re-worked like for the other `image*` tags: https://github.com/DataDog/datadog-agent/pull/15899/commits/d19363746c354884fee4f66737e6fe4ffd0a5c05